netket/exact.py

```diff
@@ -18,129 +18,109 @@
 from scipy.sparse.linalg import LinearOperator, bicgstab
 
 from . import _core
+from .operator import AbstractOperator
 from ._exact_dynamics import PyExactTimePropagation
 
 
-class EdResult(object):
-    def __init__(self, eigenvalues, eigenvectors):
-        # NOTE: These conversions are required because our old C++ code stored
-        # eigenvalues and eigenvectors as Python lists :(
-        self._eigenvalues = eigenvalues.tolist()
-        self._eigenvectors = (
-            [np.asarray(eigenvectors[:, i]) for i in range(eigenvectors.shape[1])]
-            if eigenvectors is not None
-            else []
-        )
-
-    @property
-    def eigenvalues(self):
-        r"""Eigenvalues of the Hamiltonian."""
-        return self._eigenvalues
-
-    @property
-    def eigenvectors(self):
-        r"""Eigenvectors of the Hamiltonian."""
-        return self._eigenvectors
-
-    def mean(self, operator, which):
-        import numpy
-
-        x = self._eigenvectors[which]
-
-        return numpy.vdot(x, operator(x))
-
-
 def lanczos_ed(
-    operator,
-    matrix_free=False,
-    first_n=1,
-    max_iter=1000,
-    seed=None,
-    precision=1e-14,
-    compute_eigenvectors=False,
+    operator: AbstractOperator,
+    *,
```

Comment:
what is this?

Reply:
Ah, sorry, I did not mention this change: It makes the options following in the argument list [keyword-only](https://python-3-for-scientists.readthedocs.io/en/latest/python3_advanced.html#keyword-only-arguments).

Personally, I prefer this for methods such as this, as it forces them to be called like
```python
>>> lanczos_ed(op, k=4, compute_eigenvectors=True, matrix_free=True)
...
```
instead of 
```python
>>> lanczos_ed(op, 4, True, True)
TypeError: lanczos_ed() takes 1 positional argument but 4 were given
```